src/clients/c++/request_common.cc

```diff
@@ -204,6 +205,7 @@ InputImpl::SetRaw(const uint8_t* input, size_t input_byte_size)
 
   bufs_.push_back(input);
   buf_byte_sizes_.push_back(input_byte_size);
+  use_shm_ = false;
```

Comment:
Need better checking around use_shm_. It needs to be a tri-state. Initially it is "NONE". Once SetRaw is called change it to RAW and make sure that there are no calls to SetSharedMemory. Once SetSharedMemory is called set to "SHARED_MEMORY" and make sure no calls to SetRaw or SetSharedMemory (can only be one call to that). There is a Reset() function that should set it back to "NONE". Also should update the documentation on the Reset() function to indicate this.